### PR TITLE
Enable folded syntax, several fixes to current operator syntax

### PR DESCRIPTION
**Pull request by ehildenb**

### CODE REVIEW

**Comment by hjorthjort** [comment]
This closes #34

**Comment by ehildenb** [comment]
@hjorthjort I've added tests that the `ite` operator is lazy in both branches, but strict in the condition. I've also added tests that `select` is strict in condition and both branches.

To do this, I needed to change `test.md` up a bit. That file should be re-reviewed. `wasm.md` didn't change again, except that I now force parens around `unrechable`.

File: test.md [hunk 1]
```diff
@@ -16,13 +16,24 @@ Assertions
 These assertions will check the supplied property, and then clear that state from the configuration.
 In this way, tests can be written as a serious of setup, execute, assert cycles which leaves the configuration empty on success.
 
+We'll make `Assertion` a subsort of `Instr`, so that we can consume them with `trap` statements.
+
```
Comment (hjorthjort) [inline test.md]: If you want you can add a short sentence on why we want to consume traps the way we do here, took me a bit of thinking to see that it's needed for the select tests.

Comment (ehildenb) [reply test.md]: I added some more documentation of this.

**Pull request merged**